engine/access/rest/converter.go

```diff
@@ -155,80 +304,166 @@ func toTransaction(tx *generated.TransactionsBody) (flow.TransactionBody, error)
 	}, nil
 }
 
-// Response section - converting flow models to response models.
+func toScriptArgs(script generated.ScriptsBody) ([][]byte, error) {
+	// todo(sideninja) validate
+	args := make([][]byte, len(script.Arguments))
+	for i, a := range script.Arguments {
+		arg, err := fromBase64(a)
+		if err != nil {
+			return nil, fmt.Errorf("invalid script encoding")
+		}
+		args[i] = arg
+	}
+	return args, nil
+}
+
+func toScriptSource(script generated.ScriptsBody) ([]byte, error) {
+	source, err := fromBase64(script.Script)
+	if err != nil {
+		return nil, fmt.Errorf("invalid script source encoding")
+	}
+	return source, nil
+}
+
+// Response section converting flow models to response models
+// all methods names in this section are appended with 'response'
+//
 
 func proposalKeyResponse(key *flow.ProposalKey) *generated.ProposalKey {
+
 	return &generated.ProposalKey{
 		Address:        key.Address.String(),
-		KeyIndex:       int32(key.KeyIndex),
-		SequenceNumber: int32(key.SequenceNumber),
+		KeyIndex:       fromUint64(key.KeyIndex),
+		SequenceNumber: fromUint64(key.SequenceNumber),
 	}
 }
 
 func transactionSignatureResponse(signatures []flow.TransactionSignature) []generated.TransactionSignature {
 	sigs := make([]generated.TransactionSignature, len(signatures))
-	for _, sig := range signatures {
-		sigs = append(sigs,
-			generated.TransactionSignature{
-				Address:     sig.Address.String(),
-				SignerIndex: int32(sig.SignerIndex),
-				KeyIndex:    int32(sig.KeyIndex),
-				Signature:   string(sig.Signature),
-			},
-		)
+	for i, sig := range signatures {
+		sigs[i] = generated.TransactionSignature{
+			Address:     sig.Address.String(),
+			SignerIndex: fmt.Sprintf("%d", sig.SignerIndex),
+			KeyIndex:    fromUint64(sig.KeyIndex),
+			Signature:   toBase64(sig.Signature),
+		}
 	}
 
 	return sigs
 }
 
-func transactionResponse(tx *flow.TransactionBody) *generated.Transaction {
+func transactionResponse(tx *flow.TransactionBody, txr *access.TransactionResult, link LinkGenerator, expands map[string]bool) *generated.Transaction {
 	var args []string
 	for _, arg := range tx.Arguments {
-		args = append(args, string(arg))
+		args = append(args, toBase64(arg))
 	}
 
 	var auths []string
 	for _, auth := range tx.Authorizers {
 		auths = append(auths, auth.String())
 	}
 
-	return &generated.Transaction{
+	var expandable = new(generated.TransactionExpandable)
+	var result *generated.TransactionResult
+	// if transaction result is provided then add that to the response, else add the result link to the expandable
+	if txr != nil {
+		result = transactionResultResponse(txr, tx.ID(), link)
+	} else {
+		resultLink, _ := link.TransactionResultLink(tx.ID())
```

Comment:
I feel like this error is not possible in the implementation of `TransactionResultLink ` -> `link()` since we statically define routes and for sure provide valid IDs. I'm not sure if a better way would be to remove it from the return signature. We have a bit of disagreement. Maybe another opinion can be good. Honestly I'm fine either way but would like to know why.

Reply:
If we change `TransactionResultLink` in the future to return an error under a different condition, we may not think to look for all the places in the code that call it to make sure they're handling the error properly.

I'd say, if `TransactionResultLink` returns an error, we should handle it. Since it's really a development time error instead of a runtime error, so we could just return an empty string in the case that the route isn't defined.